src/uu/du/src/du.rs

```diff
@@ -361,126 +384,187 @@ fn convert_size_other(size: u64, _multiplier: u64, block_size: u64) -> String {
+                .long("count-links")
+                .help("count sizes many times if hard linked")
         )
-        // In main
-        .optflag(
-            "S",
-            "separate-dirs",
-            "do not include size of subdirectories",
+        // .arg(
+        //     Arg::with_name("dereference")
+        //         .short("L")
+        //         .long("dereference")
+        //         .help("dereference all symbolic links")
+        // )
+        // .arg(
+        //     Arg::with_name("no-dereference")
+        //         .short("P")
+        //         .long("no-dereference")
+        //         .help("don't follow any symbolic links (this is the default)")
+        // )
+        .arg(
+            Arg::with_name(options::BLOCK_SIZE_1M)
+                .short("m")
+                .help("like --block-size=1M")
         )
-        // In main
-        .optflag("s", "summarize", "display only a total for each argument")
-        // // In task
-        // opts.optflag("x", "one-file-system", "skip directories on different file systems"),
-        // // In task
-        // opts.optopt("X", "exclude-from", "exclude files that match any pattern in FILE", "FILE"),
-        // // In task
-        // opts.optopt("", "exclude", "exclude files that match PATTERN", "PATTERN"),
-        // In main
-        .optopt(
-            "d",
-            "max-depth",
-            "print the total for a directory (or file, with --all)
-            only if it is N or fewer levels below the command
-            line argument;  --max-depth=0 is the same as --summarize",
-            "N",
+        .arg(
+            Arg::with_name(options::NULL)
+                .short("0")
+                .long("null")
+                .help("end each output line with 0 byte rather than newline")
         )
-        // In main
-        .optflagopt(
-            "",
-            "time",
-            "show time of the last modification of any file in the
-            directory, or any of its subdirectories.  If WORD is given, show time as WORD instead
-            of modification time: atime, access, use, ctime or status",
-            "WORD",
+        .arg(
+            Arg::with_name(options::SEPARATE_DIRS)
+                .short("S")
+                .long("separate-dirs")
+                .help("do not include size of subdirectories")
         )
-        // In main
-        .optopt(
-            "",
-            "time-style",
-            "show times using style STYLE:
-            full-iso, long-iso, iso, +FORMAT FORMAT is interpreted like 'date'",
-            "STYLE",
+        .arg(
+            Arg::with_name(options::SUMMARIZE)
+                .short("s")
+                .long("summarize")
+                .help("display only a total for each argument")
         )
-        .parse(args);
+        .arg(
+            Arg::with_name(options::SI)
+                .long(options::SI)
+                .help("like -h, but use powers of 1000 not 1024")
+        )
```

Comment:
when you get a chance, could you please open good first issues for these three options? thanks

if people are looking to new bugs to contribute